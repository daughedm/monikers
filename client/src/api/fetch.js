export const makeFetch = async (url, options) => {
  try {
    const response = await fetch(url, options);



//jgjh



    if (!response.ok) {
      throw new Error(`${response.status}`);
    }
    return await response.json();
  } catch (error) {
    throw new Error(`Network request failed. (error: ${error.message})`);
  }
};
