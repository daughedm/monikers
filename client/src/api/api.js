import { makeFetch } from './fetch';

export const getCards = async () => {
  const url = '/api/v1/cards';
  return await makeFetch(url);
};
