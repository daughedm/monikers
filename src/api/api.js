import { makeFetch } from './fetch';

export const getCards = async () => {
  const url = 'http://localhost:3001/api/v1/cards';
  return await makeFetch(url);
};
