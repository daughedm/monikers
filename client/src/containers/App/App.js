import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import SplashPage from '../../components/SplashPage/SplashPage';
import Play from '../../components/Play/Play';
import Setup from '../../components/Setup/Setup';
import Round from '../../components/Round/Round';
import Next from '../../components/Next/Next';
import Instructions from '../../components/Instructions/Instructions';
import Loading from '../../components/Loading/Loading';
import Finish from '../../components/Finish/Finish';
import indexedDB from '../../indexedBD';
import * as api from '../../api/api';
import cardData from '../../data/cards';
import './App.css';

class App extends Component {
  async componentDidMount() {
    await this.cardsPGtoIDB();
  }

  cardsPGtoIDB = async () => {
    if (navigator.onLine) {
      // const cardsPG = await api.getCards();
      indexedDB.allCards.clear();
      indexedDB.allCards.bulkAdd(cardData);
    }
  };

  render() {
    return (
      <div>
        <Route path="/" exact={true} component={SplashPage} />
        <Route path="/instructions" exact={true} component={Instructions} />
        <Route path="/setup" exact={true} component={Setup} />
        <Route path="/play" exact={true} component={Play} />
        <Route path="/loading" exact={true} component={Loading} />
        <Route path="/round" exact={true} component={Round} />
        <Route path="/next" exact={true} component={Next} />
        <Route path="/finish" exact={true} component={Finish} />
      </div>
    );
  }
}

export default App;
