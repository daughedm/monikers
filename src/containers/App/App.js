import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as actions from '../../actions/index';
import Setup from '../../components/Setup/Setup';
import './App.css';

export class App extends Component {
  componentDidMount() {}

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Monikers</h1>
        </header>
        <Setup />
      </div>
    );
  }
}

export const mapStateToProps = state => ({});

export const mapDispatchToProps = dispatch => ({});

App.propTypes = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
