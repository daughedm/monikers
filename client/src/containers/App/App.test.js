/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';
import App from './app';
import * as api from '../../api/api';
jest.mock('../../api/api.js');

describe('app', () => {
  let app;
  let mockProps;

  beforeEach(() => {
    mockProps = {};
    app = shallow(<App {...mockProps} />, { disableLifecycleMethods: true });
  });

  it('matches the snapshot', () => {
    expect(app).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls cardsPGtoIDB', async () => {
      const cardsPGtoIDB = (app.instance().cardsPGtoIDB = jest.fn());

      await app.instance().componentDidMount();

      expect(cardsPGtoIDB).toHaveBeenCalledTimes(1);
    });
  });

  describe('cardsPGtoIDB', () => {
    it('calls getCards if navigator.onLine is true', async () => {
      await app.instance().cardsPGtoIDB();

      expect(api.getCards).toHaveBeenCalledTimes(1);
    });

    it.skip('calls indexedDB', async () => {
      const indexedDB = { allCards: { clear: jest.fn() } };

      await app.instance().cardsPGtoIDB();

      expect(indexedDB.allCards.clear).toHaveBeenCalledTimes(1);

      expect();
    });
  });
});
