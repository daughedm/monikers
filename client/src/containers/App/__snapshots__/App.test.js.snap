// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/instructions"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/setup"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/play"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/loading"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/round"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/next"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/finish"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/instructions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/setup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/play",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/round",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/next",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/finish",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/instructions"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/setup"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/play"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/loading"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/round"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/next"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/finish"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/instructions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/setup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/play",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/loading",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/round",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/next",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/finish",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
